plugin/handlergen.go

```diff
@@ -619,12 +630,13 @@ func (p *OrmPlugin) generateStrictUpdateHandler(message *generator.Descriptor) {
 		p.P(`return nil, err`)
 		p.P(`}`)
 	}
+	p.generateBeforeHookCall(ormable, "StrictUpdateSave")
+	p.P(`if err = db.Set("gorm:association_autoupdate", false).Set("gorm:association_autocreate", false).Set("gorm:association_save_reference", false).Save(&ormObj).Error; err != nil {`)
```

Comment:
Is this mostly to lock the table? Is this the clearest way to do that? It seems redundant (the update of the base object) with the db.Save below.

Reply:
The thing is that we want to save all associations after updating the base object and deleting all associated child objects. I don't see a concise, clear way of doing this with GORM for now. We could save associations one-by-one, and that was something I was trying to achieve in
 my previous commit, but the difficulty is that we would need to save all associations(along with non-proto(included) ones), and this is a non-trivial, ambiguous thing to implement from the plugin perspective. So I ended up with the current solution.

Reply:
Is there an option to just 'lock' the table or entry somehow, or would that be no more clear or efficient than this update?

Reply:
After thinking a bit on this I realized that we can lock a row while performing count operation. SELECT FOR UPDATE is used to do this.